projects/ngneat/hotkeys/src/lib/hotkeys.directive.ts

```diff
@@ -1,71 +1,71 @@
 import { mergeAll } from 'rxjs/operators';
 
-import { AllowInElement, Hotkey, HotkeysService } from './hotkeys.service';
+import { Hotkey, HotkeysService, Options as ServiceOptions } from './hotkeys.service';
 import { coerceArray } from './utils/array';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 
-interface Options {
-  trigger: 'keydown' | 'keyup';
-  allowIn: AllowInElement[];
-  showInHelpMenu: boolean;
-  preventDefault: boolean;
-}
+type Options = Omit<ServiceOptions, 'group' | 'element' | 'description'>;
 
 @Directive({
-  selector: '[hotkeys]'
+  standalone: true,
+  selector: '[hotkeys]',
 })
-export class HotkeysDirective implements OnChanges, OnDestroy {
+export class HotkeysDirective implements OnChanges {
+  private destroyRef = inject(DestroyRef);
+  private hotkeysService = inject(HotkeysService);
+  private elementRef = inject(ElementRef);
   private subscription: Subscription;
 
-  constructor(private hotkeysService: HotkeysService, private elementRef: ElementRef) {}
-
-  @Input() hotkeys: string;
-  @Input() isSequence: boolean = false;
-  @Input() hotkeysGroup: string;
-  @Input() hotkeysOptions: Partial<Options> = {};
-  @Input() hotkeysDescription: string;
+  hotkeys = input<string>();
+  isSequence = input(false);
+  hotkeysGroup = input<string>();
+  hotkeysOptions = input<Partial<Options>>({});
+  hotkeysDescription = input<string>();
+  @Output() hotkey = new EventEmitter<KeyboardEvent | Hotkey>();
 
-  @Output()
-  hotkey = new EventEmitter<KeyboardEvent | Hotkey>();
+  private _hotkey = computed(() => ({
+    keys: this.hotkeys(),
+    group: this.hotkeysGroup(),
+    description: this.hotkeysDescription(),
+    ...this.hotkeysOptions(),
+  }));
 
   ngOnChanges(changes: SimpleChanges): void {
-    this.deleteHotkeys();
-    if (!this.hotkeys) {
-      return;
-    }
-
-    const hotkey: Hotkey = {
-      keys: this.hotkeys,
-      group: this.hotkeysGroup,
-      description: this.hotkeysDescription,
-      ...this.hotkeysOptions
-    };
-
-    this.setHotkeys(hotkey);
-  }
-
-  ngOnDestroy() {
-    this.deleteHotkeys();
-  }
-
-  private deleteHotkeys() {
     if (this.subscription) {
       this.subscription.unsubscribe();
     }
     this.subscription = null;
+
+    if (!this.hotkeys) {
+      return;
+    }
+
+    this.setHotkeys(this._hotkey());
   }
 
   private setHotkeys(hotkeys: Hotkey | Hotkey[]) {
     const coercedHotkeys = coerceArray(hotkeys);
     this.subscription = merge(
-      coercedHotkeys.map(hotkey => {
-        return this.isSequence
+      coercedHotkeys.map((hotkey) => {
+        return this.isSequence()
           ? this.hotkeysService.addSequenceShortcut({ ...hotkey, element: this.elementRef.nativeElement })
           : this.hotkeysService.addShortcut({ ...hotkey, element: this.elementRef.nativeElement });
-      })
+      }),
     )
-      .pipe(mergeAll())
-      .subscribe(e => this.hotkey.next(e));
+      .pipe(takeUntilDestroyed(this.destroyRef), mergeAll())
```

Comment:
why you need this change? you already have the subscription and the unsubscribe call